addons/http_routing/models/ir_http.py

```diff
@@ -331,17 +331,14 @@ def _get_translation_frontend_modules_name(cls):
         """
         return ['web']
 
-    bots = "bot|crawl|slurp|spider|curl|wget|facebookexternalhit".split("|")
+    bots = "bot|crawl|slurp|spider|curl|wget|facebookexternalhit|whatsapp|trendsmapresolver|pinterest|instagram".split("|")
 
     @classmethod
     def is_a_bot(cls):
         # We don't use regexp and ustr voluntarily
         # timeit has been done to check the optimum method
-        user_agent = request.httprequest.environ.get('HTTP_USER_AGENT', '').lower()
-        try:
-            return any(bot in user_agent for bot in cls.bots)
-        except UnicodeDecodeError:
-            return any(bot in user_agent.encode('ascii', 'ignore') for bot in cls.bots)
+        user_agent = request.httprequest.user_agent.string.lower()
```

Comment:
Out of curiosity, why this change?
(and also about the try/catch)

Reply:
changed ;)